examples/bri-3/src/bri/merkleTree/agents/merkleTree.agent.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import MerkleTree from 'merkletreejs';
+import { v4 } from 'uuid';
+import { BpiMerkleTree } from '../models/bpiMerkleTree';
+import { MerkleTreeStorageAgent } from './merkleTreeStorage.agent';
+import * as crypto from 'crypto';
+
+@Injectable()
+export class MerkleTreeAgent {
+  constructor(private storageAgent: MerkleTreeStorageAgent) {}
+  public createNewMerkleTree(
+    leaves: string[],
+    hashFunction: string,
```

Comment:
should we validate hash function or only support some, since this can be any string here and no control what is sent, maybe adding enum or whatever validation

but up to you

Reply:
added a TODO